code/modules/vehicles/secway.dm

```diff
@@ -47,23 +43,35 @@
 				if(obj_integrity == max_integrity)
 					to_chat(user, "<span class='notice'>It looks to be fully repaired now.</span>")
 		return TRUE
+
+	if(istype(W, /obj/item/reagent_containers/food/snacks/grown/banana))
+		if(!user.transferItemToLoc(W, src))
+			return
+		// ignore the occupants because they're presumably too distracted to notice the guy stuffing fruit into their vehicle's exhaust. do segways have exhausts? they do now!
+		user.visible_message("<span class='warning'>[user] begins stuffing [W] into [src]'s tailpipe.</span>", "<span class='warning'>You begin stuffing [W] into [src]'s tailpipe...</span>", ignored_mobs = occupants)
+		if(do_after(user, 30, TRUE, target = src))
+			user.visible_message("<span class='warning'>[user] stuffs [W] into [src]'s tailpipe.</span>", "<span class='warning'>You stuff [W] into [src]'s tailpipe.</span>", ignored_mobs = occupants)
```

Comment:
actually rethinking this, move the transferitemtoloc() above here instead, it'll handle the forceMove() as well.